### ISSUE TITLE
Custom shortcuts (right side column items) not populating recent progams list

### ISSUE THREAD

**Issue by klepp0906** [Open-Shell/Open-Shell-Menu#348]
So due to something unrelated, I just cleared my recent items list.  I'm unsure if I never noticed this - or behavior changed.

It seems that anything I run from my right panel (shortcuts I added via customize start menu) will not populate my recent programs list.

I have everything from windows apps (which I realize has a setting to display them or otherwise) I have portable programs that I have stashed away in various locations, and I have windows built in items like event viewer etc.

no matter what I run, it will not populate the recent programs list.

If i run for example, a command prompt from the customized list, I get no item in the list.   However if i go into programs and navigate to the start menu shortcut and run it - it displays fine.  

I swear this hasnt always been the case.   Is something busted on my end?

**Comment by NebelNidas** [comment]
First of all, what do you mean with "recent items list"? Is it area one or two?
![image](https://user-images.githubusercontent.com/48808497/83953779-8e2fe480-a843-11ea-82a2-5e44d390bf3a.png)
(Sorry, my start menu is German and I'm too lazy to change it, but you should be able to make it out. 😅)

If it's area one you are talking about, try my fix suggested in https://github.com/Open-Shell/Open-Shell-Menu/issues/363#issuecomment-637376854.
If it's area two, have you enabled recent files in the Windows start menu settings? Because turning this off will also affect Open Shell.
![image](https://user-images.githubusercontent.com/48808497/83953949-18c51380-a845-11ea-9131-58b31bddae79.png)

**Comment by klepp0906** [comment]
its area 1, area 2 populates with my recent documents etc fine.  Area 1 doesnt populate if an item is ran from a custom entry added via the "customize start menu" tab in open-shell settings. 

if that same item is ran via the natural menu item location it populates fine (in area 1).

I'll look into your fix.   Thank you for the reply.

**Comment by NebelNidas** [comment]
Hmm, my fix is intended for people whose frequently used items list doesn't populate at all. If only programs launched from the right hand side don't show up, it would be an Open Shell issue, indeed...

**Comment by NebelNidas** [comment]
I just tried it myself, and it doesn't work on my end, either. But are you sure this isn't the intended behavior? I think it would be annoying if programs already pinned to the right hand side also showed up in my frequently used programs list. @coddec, @ge0rdi, @Ibuprophen, do you know if this is the correct behavior?

>I swear this hasnt always been the case.

In the meantime, until the mods tell us how it really is intended to work, can you check with older versions of Open Shell? And report it back, if some version really behaves like you thought? Just to be sure, maybe it was just your memories playing a trick on you. 😉

**Comment by klepp0906** [comment]
it could have been (my memory playing tricks) however if it never worked, im of the mindset it would be nice if it did.

for example if i want to use services.msc i have to navigate all the way through the control panel > administrative tools > services hierarchy in order for it populate the list.

meanwhile, I have a shortcut pointing to that same exact file `%SystemRoot%\System32\services.msc` which if executed from the right pane, does not.  I have it there for convenience but not having what ive recently used from that pane tracked, removes some of the convenience.

**Comment by ge0rdi** [comment]
@klepp0906 
Go to `Open-shell` settings and make sure you have `Recent programs` selected under `Main Menu -> Show recent or frequent programs`.

**Comment by klepp0906** [comment]
i imagine if this wasnt selected it wouldnt be showing any recent programs at all.  But to be thorough...
![image](https://user-images.githubusercontent.com/10392920/84196690-93bd4280-aa6e-11ea-80a6-23732252c04f.png)

**Comment by NebelNidas** [comment]
That's right, @klepp0906. So in conclusion it seems like it just was never intended for right-pinned programs to show up in the recent programs list, maybe it was just a bug on your side. 🤷‍♀️ If you really want this as an official feature, you can create a new issue Feature Request, this way it may be added some day.

**Comment by Galane** [comment]
![startmenu](https://user-images.githubusercontent.com/31779300/109569818-48d4d680-7aa6-11eb-8a24-2db058c5da9d.jpg)
